### PR TITLE
fix: Helm configuration fix

### PR DESCRIPTION
**Pull request by james-milligan**
## This PR

- fixes helm configuration
- adds hardcoded rbac prefix to prevent helm breaking the referecne in `webhooks/pod_webhook.go`

### Related Issues

Fixes https://github.com/open-feature/open-feature-operator/issues/199

### Follow-up Tasks
Hardcoded rbac prefix value should be removed in favour of making the `webhooks/pod_webhook.go` value configurable.
https://github.com/open-feature/open-feature-operator/issues/202

### How to test

follow the README instructions for helm deployment, then deploy the end-to-end example, there should be no errors and app should behave correctly.

### CODE REVIEW

File: chart/templates/deployment.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ spec:
         - /manager
         env:
         - name: FLAGD_VERSION
-          value: INPUT_FLAGD_VERSION
+          value: latest
```
Comment (toddbaert) [inline chart/templates/deployment.yaml:38]: Is this change intended

There seems to be a shell script for `sed`ing this in `hack/`.

**Review by toddbaert — APPROVED** [review]
in `main`, If I install via the helm command included here, at the point I deploy a workload (and a webhook is called) I get:

```
Error from server (InternalError): error when creating "end-to-end.yaml": Internal error occurred: failed calling webhook "validate.featureflagconfiguration.openfeature.dev": failed to call webhook: Post "https://ofo-webhook-service.open-feature-operator-system.svc:443/validate-v1alpha1-featureflagconfiguration?timeout=10s": x509: certificate is valid for ofo-$(SERVICE_NAME).$(SERVICE_NAMESPACE).svc, ofo-$(SERVICE_NAME).$(SERVICE_NAMESPACE).svc.cluster.local, not ofo-webhook-service.open-feature-operator-system.svc
```
It seems like the transformation issue we had previously remains in the case of helm.

This PR resolves this issue for me. I think eventually we should replace all `kustomize` usage` with `helm` - having both seems a bit wonky - but that's just my opinion, maybe there's a good reason we should keep both.

File: chart/templates/deployment.yaml (hunk 1 above)
Comment (AlexsJones) [reply chart/templates/deployment.yaml:38]: Please don't do this, that value is set intentionally

**Review by AlexsJones — CHANGES_REQUESTED** [review]
Let's meet to discuss, hard coding this doesn't feel right

**Comment by james-milligan** [comment]
closing in favour of  #206
